### ISSUE TITLE
Textext error on debian 9

### ISSUE THREAD

**Issue by 1koopa** [textext/textext#43]
I have just installed textext for linux (debian 9) following this guide:
https://github.com/textext/textext/wiki/Installation-specific-instructions-for-Linux-systems
and am getting this error when I try and run it in Inkscape. 

On the window "Enter latex formula - TexText" if I click on Preview option:
Command pdflatex /tmp/tmpaKlhHw/tmp.tex -interaction=nonstopmode -halt-on-error failed: [Errno 2] No such file or directory 

Click on Save option:
Command pdflatex /tmp/tmpCodFV5/tmp.tex -interaction=nonstopmode -halt-on-error failed: [Errno 2] No such file or directory
Traceback (most recent call last):
  File "/home/k/.config/inkscape/extensions/asktext.py", line 599, in cb_ok
    self._texcmd_cbox.get_active_text().lower())
  File "textext.py", line 292, in callback
    tex_cmd, original_scale=current_scale)
  File "textext.py", line 393, in do_convert
    new_svg_ele = converter.convert(text, preamble_file, scale_factor, tex_command)
  File "textext.py", line 820, in convert
    self.tex_to_pdf(tex_command, latex_text, preamble_file)
  File "textext.py", line 766, in tex_to_pdf
    raise RuntimeError(latest_message())
RuntimeError: Command pdflatex /tmp/tmpCodFV5/tmp.tex -interaction=nonstopmode -halt-on-error failed: [Errno 2] No such file or directory

I have Texlive installed. I use this guide:
https://tex.stackexchange.com/questions/1092/how-to-install-vanilla-texlive-on-debian-or-ubuntu

More specific, I installed "vanilla" Texlive.

**Comment by jcwinkler** [comment]
Most likely the pdflatex command is not found (the error message would differ if the tex file is not found by pdflatex). 

Please open a terminal and enter the command ```pdflatex``` followed by pressing Return. Is the command successfully executed ("This is pdfTeX, Version ... ")?

**Comment by 1koopa** [comment]
It is show me this:
This is pdfTeX, Version 3.14159265-2.6-1.40.19 (TeX Live 2018) (preloaded format=pdflatex)
 restricted \write18 enabled.

**Comment by jcwinkler** [comment]
So pdflatex is on the system path but not found when called from ```subprocess.Popen``` from TexText.

Similar to the problem described here: https://stackoverflow.com/questions/4230926/pdflatex-in-a-python-subprocess-on-mac

But the solution given there (absolute path) is not applicable to TexText, of course.  Does anyone else has an idea?

**Comment by sizmailov** [comment]
Yes, it's puzzling. pdflatex works in terminal and doesn't in TexText.

What can lead to this? I can imagine two possibilities:
1) different values of $PATH in terminal and in inkscape/TexText
2) pdflatex is an alias, so it's unreachable from subprocess 


@1koopa can you check environment variables in TexText by changing couple of lines in textext.py by replacing 

https://github.com/textext/textext/blob/0164a57dc65bb8f2024e315ab7a0769afe0fe933/textext.py#L610-L616

with 

```python
        except OSError, err:
            add_log_message(str(os.environ) + "\nCommand %s failed: %s" % (' '.join(cmd), err), LOG_LEVEL_DEBUG)
            raise RuntimeError(latest_message())

        if ok_return_value is not None and p.returncode != ok_return_value:
            add_log_message(str(os.environ) + "\nCommand %s failed (code %d): %s" % (' '.join(cmd), p.returncode, out + err),
                            LOG_LEVEL_DEBUG)
```

Please also paste output of `env`, `alias`, `which pdflatex` commands from terminal. 

Wipe any confidential information from `env` output if there are any.

**Comment by 1koopa** [comment]
Command env:
XDG_MENU_PREFIX=gnome-
LANG=en_US.UTF-8
GDM_LANG=en_US.UTF-8
DISPLAY=:1
COLORTERM=truecolor
DESKTOP_AUTOSTART_ID=xxxxxxxxxxxxxxxxxxxxxx
USERNAME=k
XDG_VTNR=2
SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
XDG_SESSION_ID=4
USER=k
DESKTOP_SESSION=default
QT_QPA_PLATFORMTHEME=qgnomeplatform
PWD=/home/k
HOME=/home/k
JOURNAL_STREAM=8:18651
SSH_AGENT_PID=904
QT_ACCESSIBILITY=1
XDG_SESSION_TYPE=x11
XDG_DATA_DIRS=/usr/share/gnome:/usr/local/share/:/usr/share/
XDG_SESSION_DESKTOP=default
GTK_MODULES=gail:atk-bridge
WINDOWPATH=2
TERM=xterm-256color
SHELL=/bin/bash
VTE_VERSION=4601
XDG_CURRENT_DESKTOP=GNOME
GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1
QT_LINUX_ACCESSIBILITY_ALWAYS_ON=1
SHLVL=1
XDG_SEAT=seat0
MANPATH=/usr/local/texlive/2018/texmf-dist/doc/man:
WINDOWID=xxxxxxx
GDMSESSION=default
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
LOGNAME=k
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
XDG_RUNTIME_DIR=/run/user/1000
XAUTHORITY=/run/user/1000/gdm/Xauthority
PATH=/usr/local/texlive/2018/bin/x86_64-linux:/home/k/anaconda3/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
INFOPATH=/usr/local/texlive/2018/texmf-dist/doc/info:
SESSION_MANAGER=local/debian:@/tmp/.ICE-unix/849,unix/debian:/tmp/.ICE-unix/849
_=/usr/bin/env

Comand alias:
alias ls='ls --color=auto'

Comand which pdflatex:
/usr/local/texlive/2018/bin/x86_64-linux/pdflatex

On the window "Enter latex formula - TexText" (I changed couple of lines in textext.py):
{'GNOME_DESKTOP_SESSION_ID': 'this-is-deprecated', 'GJS_DEBUG_OUTPUT': 'stderr', 'XDG_CURRENT_DESKTOP': 'GNOME', 'XDG_SESSION_TYPE': 'x11', 'LOGNAME': 'k', 'XDG_SEAT': 'seat0', 'HOME': '/home/k', 'XDG_VTNR': '2', 'PATH': '/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games', 'DISPLAY': ':1', 'SSH_AGENT_PID': '904', 'LANG': 'en_US.UTF-8', 'SHELL': '/bin/bash', 'XAUTHORITY': '/run/user/1000/gdm/Xauthority', 'JOURNAL_STREAM': '8:18651', 'SESSION_MANAGER': 'local/debian:@/tmp/.ICE-unix/849,unix/debian:/tmp/.ICE-unix/849', 'XDG_DATA_DIRS': '/usr/share/gnome:/usr/local/share/:/usr/share/', 'QT_LINUX_ACCESSIBILITY_ALWAYS_ON': '1', 'QT_QPA_PLATFORMTHEME': 'qgnomeplatform', 'QT_ACCESSIBILITY': '1', 'WINDOWPATH': '2', 'GIO_LAUNCHED_DESKTOP_FILE_PID': '4359', 'GPG_AGENT_INFO': '/run/user/1000/gnupg/S.gpg-agent:0:1', 'USERNAME': 'k', 'XDG_SESSION_DESKTOP': 'default', 'GIO_LAUNCHED_DESKTOP_FILE': '/usr/share/applications/inkscape.desktop', 'XDG_RUNTIME_DIR': '/run/user/1000', 'PYTHONPATH': '/usr/share/inkscape/extensions', 'SSH_AUTH_SOCK': '/run/user/1000/keyring/ssh', 'GDMSESSION': 'default', 'XDG_SESSION_ID': '4', 'DBUS_SESSION_BUS_ADDRESS': 'unix:path=/run/user/1000/bus', 'PACKAGE_LOCALE_DIR': '/usr/share/locale', 'DESKTOP_SESSION': 'default', 'GTK_MODULES': 'gail:atk-bridge', 'GDM_LANG': 'en_US.UTF-8', 'PWD': '/home/k', 'XDG_MENU_PREFIX': 'gnome-', 'GJS_DEBUG_TOPICS': 'JS ERROR;JS LOG', 'USER': 'k'}
Command pdflatex /tmp/tmpad7lj1/tmp.tex -interaction=nonstopmode -halt-on-error failed: [Errno 2] No such file or directory

**Comment by sizmailov** [comment]
I assume PATH variable is extended in your `~/.bashrc`. Thus you are able to run pdflatex in terminal. 

At the same time this changes are NOT propagated to inkscape and in turn to TexText. I have tested on ubuntu but should be the case for debian too: `~/.bashrc` does not affect environment of "desktop" applications, while `/.profile` does. So the hot fix would be to copy lines related to PATH to your `~/.profile`. 

The bottom line: environment of inkscape application is different from environment of terminal shell. This is the result of rules applied to .profile/.bashrc/bash_profile/etc. in linux systems and they are quite involved (see https://askubuntu.com/a/463479/84575)

@1koopa does modification of .profile helped?

**Comment by 1koopa** [comment]
I'm newbie on Linux. How can I change the Path on /.profile ??

**Comment by sizmailov** [comment]
look for lines like 
```bash
export PATH="...:$PATH"
```
or 
```bash
PATH="...:$PATH"
export PATH
```
where "..." should contain path to pdflatex 

from the outputs you post I think that this should do the trick:

```bash
export PATH="/usr/local/texlive/2018/bin/x86_64-linux:$PATH"
```
Just append it to ".profile"

Afterwards you need to re-login to see effect.
